lib/tesla/adapter/gun.ex

```diff
@@ -101,71 +120,141 @@ if Code.ensure_loaded?(:gun) do
       request(method, url, headers, body, opts)
     end
 
-    defp request(method, url, headers, body, opts) do
-      with {pid, f_url} <- open_conn(url, opts),
-           stream <- open_stream(pid, method, f_url, headers, body, false) do
-        read_response(pid, stream, opts)
-      end
-    end
+    defp request(method, url, headers, body, opts),
+      do: do_request(method, url, headers, body, opts)
 
-    defp request_stream(method, url, headers, body, opts) do
-      with {pid, f_url} <- open_conn(url, opts),
-           stream <- open_stream(pid, method, f_url, headers, body, true) do
+    defp request_stream(method, url, headers, body, opts),
+      do: do_request(method, url, headers, body, opts)
+
+    defp do_request(method, url, headers, body, opts) do
+      uri = URI.parse(url)
+      headers = headers ++ [{"host", uri.host}]
```

Comment:
This will give wrong results when using non-standart port, no? I think instead you should just add a warning that gun <=1.3 (it's fixed in master) will append the port to the host header even if it's standart

Reply:
Adding `host` header is removed.